core/extensions/impl/crypto_extension.cpp

```diff
@@ -168,4 +174,75 @@ namespace kagome::extensions {
 
     memory_->storeBuffer(out_ptr, common::Buffer(hash));
   }
+
+  runtime::PointerSize CryptoExtension::ext_crypto_secp256k1_ecdsa_recover_v1(
+      runtime::WasmPointer sig, runtime::WasmPointer msg) {
+    static auto make_result = [&](auto &&value) {
+      boost::optional<crypto::Secp256k1UncompressedPublicKey> result = value;
+      return common::Buffer(scale::encode(result).value());
+    };
+    static auto error_result = make_result(boost::none);
```

Comment:
Why not make it inline?:
```cpp
static auto error_result = common::Buffer(boost::optional<crypto::Secp256k1UncompressedPublicKey>{boost::none}); 
```

Reply:
You can also move this variable outside function and reuse in the second method

Reply:
refactored